File: apps/api/src/app/muscle/muscle.entity.ts
import {
    Column,
    Entity,
    JoinColumn,
    ManyToOne,
    OneToMany,
    PrimaryGeneratedColumn
} from 'typeorm';
import { MuscleGroupEntity } from '../muscle-group/muscle-group.entity';
import { ExerciseMuscleEntity } from '../exercise-muscle/exercise-muscle.entity';

@Entity({ name: 'muscles', schema: 'public' })
export class MuscleEntity {
    @PrimaryGeneratedColumn('increment')
    id: number;

    @ManyToOne(() => MuscleGroupEntity, (muscleGroup) => muscleGroup.muscle)
    @JoinColumn({ name: 'muscle_group_id' })
    muscleGroup: MuscleGroupEntity;

    @Column()
    name: string;

    @OneToMany(() => ExerciseMuscleEntity, (exerciseMuscle) => exerciseMuscle.muscle)
    exerciseMuscle: ExerciseMuscleEntity[];
}

File: my-workout-api/src/main/java/com/avernet/myworkoutapi/muscle/MuscleEntity.java
package com.avernet.myworkoutapi.muscle;

import com.avernet.myworkoutapi.exercisemuscle.ExerciseMuscleEntity;
import com.avernet.myworkoutapi.musclegroup.MuscleGroupEntity;
import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.Data;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "muscles")
@Data
public class MuscleEntity {
    @Id
    Long id;

    @Column()
    String name;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "muscle_group_id")
    MuscleGroupEntity muscleGroup;


    @OneToMany(mappedBy = "muscle", fetch = FetchType.LAZY, cascade = CascadeType.ALL, orphanRemoval = true)
    List<ExerciseMuscleEntity> exerciseMuscle = new ArrayList<>();
}
